docs/core/entity/vacuum.md

```diff
@@ -3,6 +3,8 @@ title: Vacuum Entity
 sidebar_label: Vacuum
 ---
 
+Derive entity platforms from [`homeassistant.components.vacuum.VacuumEntity`](https://github.com/home-assistant/home-assistant/blob/master/homeassistant/components/vacuum/__init__.py) or [`homeassistant.components.vacuum.StateVacuumEntity`](https://github.com/home-assistant/home-assistant/blob/master/homeassistant/components/vacuum/__init__.py)
```

Comment:
`VacuumEntity` is legacy and we shouldn't link to that one. It shouldn't be used for new code.

Reply:
is there an ADR or discussion (_or something other_) about this?
If yes, than I would also add a hint to the developer docs, that `VacuumEntity` should not be used anymore.
I guess the `StateVacuumEntity` should be used instead?

Reply:
There's no ADR. We introduced the new entity class and kept the other one around to allow integrations to update in their own time. The old one is not to be used for new integrations.

Yes, `StateVacuumEntity` should be used.

Reply:
So the `VacuumEntity` should somehow be marked as deprecated and `StateVacuumEntity` should be used instead?
Is there a date till when the migrations should be done? Or should i just add a hint to this page like:
> `VacuumEntity` is deprecated and removed in future release, please use or migrate to `StateVacuumEntity`

Reply:
There's no date yet. Adding a hint sounds good. Or we remove it from the documention.

Reply:
Added a note about deprecation of `VacuumEntity` and removed link to source code